ceph_iscsi_config/common.py

```diff
@@ -482,13 +482,12 @@ def unlock(self):
 
     def _seed_rbd_config(self):
 
```

Comment:
The open/lock related changes are not related to your original issue right? They fix things like leaks.

Could you make this into a proper patchset? If this is one of your first patchsets to an upstream project here are some hints/guidelines that I use:

1. When upstream merges your PR into master, we don't want to see your git history. For example we do not want to see this:

Merge pull request #1 from zhouyh-orcadt/fix_scan_image_stuck
@zhouyh-orcadt
zhouyh-orcadt committed on Mar 20

and we want to only see the final patches. We do not want to see version 1 of a patch then a fix to that.

Instead of adding patch on top of patches you can/should also do something like a force push.

So let's say you fix some issue and make a PR. You then get a review comment asking you to add some extra code to the patch. In your local git tree branch you can revert your original patch, fix up the code, commit it. Then do

git push --force your_branch master

This will basically overwrite the old branch with the new commit and in the git history we only see the final patch which is all we care about in the main tree. The nice thing is that this will also automatically update the PR with the new code, so you do not have to do open a new PR.

2. Try to do one cleanup/fix in each patch. We currently have a mix and match of stuff going on in each patch. Later if we have a regression and need to revert, or need to cherry pick a specific patch, we can't because the changes are spread over multiple patches.

3. Add a description in the git commit, so later when someone does a "git log" they know what the patch does.

So for example for 2 and 3, you would have 3 patches.

patch 1 - fix the close/lock issues. In the patch description you could write something like:

This patch fixes an issue where the common.py is cleaning the ioctx open if a lock call fails.

patch 2- fix the error_msg issue. In the patch description you could write something like:

This patch fixes an issue where we are not checking for config commit errors and returning a error_msg in a couple places.

patch 3 - fix the RO issue. In the patch description you could write what you wrote here:

https://github.com/ceph/ceph-iscsi/pull/181#discussion_r396291475

It's a really good desrciption of the problem and how your patch solves it.

Reply:
Also, thanks for the fixing all these issues.

Let me know if you do not have time to fix up the patchset.

I can do it manually, and you will get the credit in the git commit logs, but I think due to a bug in GH it will not could in your GH stats if you care about them for work or personal reasons.